### PR TITLE
feat(fs_posix): add error code conversion

### PR DESCRIPTION
**Pull request by FASTSHIFT**
| errno      | Description                  | LVGL FS Error Code       | Note                                           |
|------------|------------------------------|-------------------------|------------------------------------------------|
| 0          | -                            | LV_FS_RES_OK            | -                                              |
| EIO        | I/O error                    | LV_FS_RES_HW_ERR        | Low level hardware error                       |
| EFAULT     | Bad address                  | LV_FS_RES_FS_ERR        | Error in the file system structure             |
| ENOENT     | No such file or directory    | LV_FS_RES_NOT_EX        | Driver, file, or directory does not exist      |
| ENOSPC     | No space left on device      | LV_FS_RES_FULL          | Disk full                                     |
| EALREADY   | Operation already in progress | LV_FS_RES_LOCKED        | The file is already opened                     |
| EACCES     | Permission denied             | LV_FS_RES_DENIED        | Access denied. Check 'fs_open' modes and write protect |
| EBUSY      | Device or resource busy       | LV_FS_RES_BUSY          | The file system now can't handle it, try later |
| ETIMEDOUT  | Connection timed out          | LV_FS_RES_TOUT          | Process timed out                              |
| ENOSYS     | Invalid system call number    | LV_FS_RES_NOT_IMP       | Requested function is not implemented           |
| ENOMEM     | Out of memory                | LV_FS_RES_OUT_OF_MEM    | Not enough memory for an internal operation     |
| EINVAL     | Invalid argument              | LV_FS_RES_INV_PARAM     | Invalid parameter among arguments               |
| default    | -                            | LV_FS_RES_UNKNOWN       | Other unknown error                            |

### Notes
- Update the [Documentation](https://github.com/lvgl/lvgl/tree/master/docs) if needed.
- Add [Examples](https://github.com/lvgl/lvgl/tree/master/examples) if relevant.
- Add [Tests](https://github.com/lvgl/lvgl/blob/master/tests/README.md) if applicable.
- If you added new options to `lv_conf_template.h` run [lv_conf_internal_gen.py](https://github.com/lvgl/lvgl/blob/master/scripts/lv_conf_internal_gen.py) and update [Kconfig](https://github.com/lvgl/lvgl/blob/master/Kconfig).
- Run `scripts/code-format.py` (`astyle v3.4.12` needs to installed by running `cd scripts; ./install_astyle.sh`) and follow the [Code Conventions](https://docs.lvgl.io/master/CODING_STYLE.html).
- Mark the Pull request as [Draft](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request) while you are working on the first version, and mark is as _Ready_ when it's ready for review.
- When changes were requested, [re-request review](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) to notify the maintainers.
- Help us to review this Pull Request! Anyone can [approve or request changes](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/approving-a-pull-request-with-required-reviews).

### CODE REVIEW

File: src/libs/fsdrv/lv_fs_posix.c [hunk 1]
```diff
@@ -327,11 +328,63 @@ static lv_fs_res_t fs_dir_close(lv_fs_drv_t * drv, void * dir_p)
...
+            return LV_FS_RES_FS_ERR;
+
+        case ENOENT: /* No such file or directory */
+            return LV_FS_RES_NOT_EX;
+
+        case ENOSPC: /* No space left on device */
+            return LV_FS_RES_FULL;
+
+        case EBUSY: /* Operation already in progress */
+            return LV_FS_RES_LOCKED;
```
Comment (XuNeo) [inline src/libs/fsdrv/lv_fs_posix.c:361]: Why not busy to busy?
EAGAIN should be handled in lv_fs_posix.c by doing again.

Comment (FASTSHIFT) [reply src/libs/fsdrv/lv_fs_posix.c:361]: I don't think it's necessary. This should be left to the upper layer to do. `lv_fs_posix` only does pure API call passthrough.

Comment (XuNeo) [reply src/libs/fsdrv/lv_fs_posix.c:361]: But no upper layer has done this. I remembered we have fixed it the `EAGAIN` issue for v8. It should be done in fs layer such as lv_fs_posix. Not all fs have EAGAIN.

I suggest remove EAGAIN and deal it internally, map EBUSY to EBUSY.

Comment (kisvegabor) [reply src/libs/fsdrv/lv_fs_posix.c:361]: According to the comment [here](https://github.com/lvgl/lvgl/blob/master/src/misc/lv_fs.h#L40) `LV_FS_RES_LOCKED` means "The file is already opened". So mapping busy to "already opened" can be misleading. In light of that I also vote for busy to busy.

Comment (FASTSHIFT) [reply src/libs/fsdrv/lv_fs_posix.c:361]: > But no upper layer has done this. I remembered we have fixed it the `EAGAIN` issue for v8. It should be done in fs layer such as lv_fs_posix. Not all fs have EAGAIN.
> 
> I suggest remove EAGAIN and deal it internally, map EBUSY to EBUSY.

`EAGAIN` does not need to be processed. This comment comes from the `write` function. It only needs to be processed when `O_NONBLOCK` is set. Our `fs_open` function does not set this flag.
```c
 *  EAGAIN
 *    Non-blocking I/O has been selected using O_NONBLOCK and the write
 *    would block.
```

Comment (FASTSHIFT) [reply src/libs/fsdrv/lv_fs_posix.c:361]: > I suggest remove EAGAIN and deal it internally, map EBUSY to EBUSY.

Done.

**Review by XuNeo — APPROVED** [review]

**Review by kisvegabor — APPROVED** [review]

**Pull request merged**